File: PracticasProgramacion/Practica_12/python/main.py
"""
Practica_12 - Matrices: A * constante, A * B.
"""

from matriz import Matriz

if __name__ == "__main__":
    a = Matriz(2, 2)
    b = Matriz(2, 2)
    a.llenar(1.0)
    b.llenar(2.0)

    print("A * 3:")
    ac = a.multiplicar_por_constante(3.0)
    ac.imprimir()

    print("\nA * B:")
    c = a.multiplicar(b)
    c.imprimir()

File: PracticasProgramacion/Practica_12/cpp/main.cpp
/**
 * Practica_12 - Matrices: A x B, A * constante, A * B
 */

#include "Matriz.h"
#include <iostream>

int main() {
    Matriz A(2, 2), B(2, 2);
    A.llenar(1.0); B.llenar(2.0);

    std::cout << "A * 3:\n";
    Matriz Ac = A.multiplicarPorConstante(3.0);
    Ac.imprimir();

    std::cout << "\nA * B:\n";
    Matriz C = A.multiplicar(B);
    C.imprimir();

    return 0;
}
